plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -244,6 +213,28 @@ public static void rootConfigGeneration() throws Exception {
         }
     }
 
+    public static void lookupBaseConfigurator(String configName) throws Exception {
+        ConfigurationAsCode configurationAsCodeObject = ConfigurationAsCode.get();
+        for (Object configuratorObject : configurationAsCodeObject.getConfigurators()) {
+            if (configuratorObject instanceof BaseConfigurator) {
+                BaseConfigurator baseConfigurator = (BaseConfigurator) configuratorObject;
+                if(configName.equals(baseConfigurator.getName())){
+                    List<Attribute> baseConfigAttributeList = baseConfigurator.getAttributes();
```

Comment:
@oleg-nenashev @casz The lookup will terminate at the configurators if we have all of the sub attributes as a part of the base configurators itself, is there scope of abstracting the way configurators are extracted from the plugins.
Could every plugin return its own configurators which would make generating the schema much easier.
For eg 
`getConfigurators(context) {
        if (context == jsonSchema) {
        }
    } would return 
`
            mailerPlugin:
                smtpHost    :
                smtpAddress :
                adminAddress:`

And the schema would then be generated following this.